### PR TITLE
Finalize shock capture and add switch for interface compressibility

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The shock capturing approach implemented used an implicit implementation of the added viscosity. This was adequate, but lead to a non-linear system which was very difficult to converge
- The interface compressibility term added was wrecking cases with surface tension

# Description of the solution

- Explicit the added viscosity term
- Finalize the formulation to remove the added viscosity in the direction of the velocity vector
- Fix the initial condition of the 3D dam break which was very discontinuous
- Add a switch to consider (or not) interface compression in the VOF cases. Default value is to disable compressibility effect. The parameter is called compressible and a default value of false.

# How Has This Been Tested?

- Some tests had to be updated slightly

# Comments
- This shock capture performs well. I think that's the end of it
- Interface compressibility came as a weird surprise. I was not expecting it to have such a dramatic effect inside droplets or within cases with surface tension effects
- I will make another PR in the following week to add this to the theory guide!

### CODE REVIEW

File: examples/multiphysics/3d-dam-break/3d-dam-break.prm [hunk 1]
```diff
@@ -12,7 +12,7 @@ subsection simulation control
   set time end                     = 6
   set time step                    = 0.001
   set adapt                        = true
-  set max cfl                      = 0.5
+  set max cfl                      = 0.8
```
Comment (hepap) [inline examples/multiphysics/3d-dam-break/3d-dam-break.prm:15]: it should be also change in the .rst of the example

File: examples/multiphysics/3d-dam-break/3d-dam-break.prm [hunk 2]
```diff
@@ -76,6 +76,11 @@ subsection initial conditions
 
   subsection VOF
     set Function expression = if (x>1.992 & z<0.55 & y>=-0.5, 1, 0)
+
+    subsection projection step
+      set enable           = true
+      set diffusion factor = 1
+    end
```
Comment (hepap) [inline examples/multiphysics/3d-dam-break/3d-dam-break.prm:83]: it should be also change in the .rst of the example

File: source/solvers/vof_assemblers.cc [hunk 3]
```diff
@@ -107,13 +104,11 @@ VOFAssemblerCore<dim>::assemble_matrix(VOFScratchData<dim> &      scratch_data,
...
           // stabilization
-          strong_jacobian_vec[q][j] += velocity * grad_phi_phase_j +
-                                       phi_phase_j * velocity_divergence -
-                                       diffusivity * laplacian_phi_phase_j;
+          strong_jacobian_vec[q][j] +=
+            velocity * grad_phi_phase_j - diffusivity * laplacian_phi_phase_j;
 
-          // DCDD shock capturing
-          if (DCDD)
-            strong_jacobian_vec[q][j] += -vdcdd * laplacian_phi_phase_j;
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:116]: Is this removed in order to add diffusivity only in the right direction, or was it a previous error to add it?

Comment (blaisb) [reply source/solvers/vof_assemblers.cc:116]: Yeah it was a bad idea to add it actually

File: doc/source/parameters/cfd/volume_of_fluid.rst [hunk 4]
```diff
@@ -78,6 +79,7 @@ The default values of the VOF parameters are given in the text box below.
     Applying viscous dissipation in one of the fluids instead of both is particularly useful when one of the fluids is air. For numerical stability, the ``kinematic viscosity`` of the air is usually increased. However, we do not want to have viscous dissipation in the air, because it would result in an unrealistic increase in its temperature. This parameter is used only if ``set heat transfer = true`` and ``set viscous dissipation = true`` in :doc:`./multiphysics`.
 
 * ``diffusivity``: value of the diffusivity (diffusion coefficient) in the transport equation of the phase fraction. Default value is ``0`` to have pure advection. Increase ``diffusivity`` to :ref:`improve wetting`.
+* ``compressible``: enables interface compression (:math:`\phi \nabla \cdot \mathbf{u}`) in the VOF equations.  This term should be kept to its default value of ``false`` except when compressible equations of state are used.
```
Comment (AmishgaAlphonius) [inline doc/source/parameters/cfd/volume_of_fluid.rst:82]:
```suggestion
* ``compressible``: enables interface compression (:math:`\phi \nabla \cdot \mathbf{u}`) in the VOF equation.  This term should be kept to its default value of ``false`` except when compressible equations of state are used.
```

**Comment by blaisb** [comment]
@hepap and @AmishgaAlphonius I applied both of your suggestions. If you are cool with it,  I'll merge after CI

File: examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-adaptive-sharpening.prm
Comment (AmishgaAlphonius) [inline examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-adaptive-sharpening.prm:1]: Just like the 3D dam-break the changes must be made in the ``.rst`` :)

File: examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-adaptive-sharpening.prm [hunk 5]
```diff
@@ -8,15 +8,15 @@ set dimension = 2
...
-  set output time    = 0.005
+  set method           = bdf2
+  set time end         = 0.75
+  set time step        = 0.001
+  set adapt            = true
+  set max cfl          = 0.8
+  set output name      = rayleigh-taylor-constant-sharpening
+  set output path      = ./output/constant/
+  set output control   = iteration
+  set output frequency = 10
```
Comment (hepap) [inline examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-adaptive-sharpening.prm:19]: This should be updated in the .rst :)

**Review by hepap — APPROVED** [review]
One last comment :D

**Review by AmishgaAlphonius — APPROVED** [review]

**Pull request merged**